packages/server-core/src/projects/project-permission/project-permission.hooks.ts

```diff
@@ -205,6 +206,41 @@ const makeRandomProjectOwner = async (context: HookContext<ProjectPermissionServ
   if (context.id && context.result) await context.service.makeRandomProjectOwnerIfNone(result[0].projectId)
 }
 
+/**
+ * Filter permissions for logged in user
+ * @param context
+ * @returns
+ */
+const filterUserPermissions = async (context: HookContext<ProjectPermissionService>) => {
+  const loggedInUser = context.params!.user!
+  if (await checkScope(loggedInUser, 'projects', 'read')) return
+
+  if (!context.params.query) {
+    throw new BadRequest(`No query in ${context.params}`)
+  }
+  context.params.query.userId = loggedInUser.id
+}
+
+/**
+ * resolve project id from name in query
+ * @param context
+ * @returns
+ */
+const resolveProjectId = async (context: HookContext<ProjectPermissionService>) => {
+  if (!context.params.query?.project) {
+    return
+  }
+
+  const projectResult = (await context.app.service(projectPath).find({
+    query: { name: context.params.query?.project, $limit: 1 }
```

Comment:
I dont see any `project` field in query in project-permission schema

Reply:
per our conversation this has been added to projectPermissionQuerySchema in order to resolve the project id 
and will be dropped from the query when the project id resolved.